src/PostgrestFilterBuilder.ts

```diff
@@ -60,8 +67,8 @@ export default class PostgrestFilterBuilder<T> extends PostgrestTransformBuilder
    * @param column  The column to filter on.
    * @param value  The value to filter with.
    */
-  eq(column: keyof T, value: T[keyof T]): this {
-    this.url.searchParams.append(`${column}`, `eq.${value}`)
+  eq<ColumnName extends string & keyof Table>(column: ColumnName, value: Table[ColumnName]): this {
```

Comment:
Looking super cool @soedirgo!

Just wondering if there are plans to make this `ColumnName` type even more helpful?
I see you're now allowing all strings plus suggesting the keys of the table – is it also possible to suggest the keys for embedded tables when the user provides the type?

[For example](https://www.typescriptlang.org/play?#code/C4TwDgpgBAyhCGwoF4oG8BQUoEsAmAXFAM7ABOOAdgOZZRkD2A7kaRTXZQK4C2R3PAEYQydHgwBuOCAH18rclVoBfDBlCQoAWUnSU6OvJKKO2YDmAAbCAva1sxBMGJE4iANoBdOsNIzHiK5OGKrq4NAA0hAgxADyAGYAPAAqADRQUSBQEAAewBCUeMRQANbRDPFQyfplIBVVAHz6mdl5BUXGdnTYAPxV7pmerfmFxQAUVPEiUACqAJRe3dhQfQAGACRomcoAdJuZcUkzDcqrS9hEyQPRQ7kjHQBKEADGDGR4iWxK6fCUIA3nZZrfbRXYgmIJFLXECeE5nZYIqBETJLfgQCQiADcag00B0UggAAVEAALZrRQ6JfHSBpAA):
```typescript
type KeysOf<T, Key extends keyof T = keyof T> = Key extends string
    ? T[Key] extends (infer U)[]
      ? `${Key}.${KeysOf<U>}`
      : T[Key] extends Record<string, any>
        ? `${Key}.${KeysOf<T[Key]>}`
        : Key
    : never;
```
(not thoroughly tested!)

Is this in scope for these changes? Or are we not going to allow users to provide nested types and have them as unknowns? (haven't dived into this deeply, so not fully informed sorry!)

Reply:
The implementation in https://github.com/bnjmnt4n/supabase-client/blob/a1bd9d86fc04ffe9fd5172667644755d96456e29/src/postgrest/PostgrestFilterBuilder.ts#L39 actually does this, but I wanted to keep this simple and fall back to `unknown` for now.

I noticed that it's *not* falling back to `unknown` though - will update the PR. Thanks for noticing!